radiation/FatesRadiationDriveMod.F90

```diff
@@ -438,11 +439,18 @@ subroutine FatesSunShadeFracs(nsites, sites,bc_in,bc_out)
                            if_notair: if (ft>0) then
                               area_frac = twostr%scelg(cl,icol)%area
                               vai = twostr%scelg(cl,icol)%sai+twostr%scelg(cl,icol)%lai
-                              nv = minloc(dlower_vai, DIM=1, MASK=(dlower_vai>vai))
+
+                              nv = GetNVegLayers(vai)
```

Comment:
This is a fix: Since the older version of dlower_vai had a non-zero first index, if vai is smaller than dlower_vai(1) but non-zero, then the mask on dlower_vai is all FALSE. So the result of minloc will be zero and there will be no looping here (which would be incorrect).

Reply:
The good part is that these are not prognostic, only diagnostics variables in this loop.